app/src/forms/proxy/service.js

```diff
@@ -0,0 +1,118 @@
+const { encryptionService } = require('../../components/encryptionService');
+const { ExternalAPI } = require('../../forms/common/models');
+
+const headerValue = (headers, key) => {
+  if (headers && key) {
+    try {
+      return headers[key.toUpperCase()] || headers[key.toLowerCase()];
+    } catch (error) {
+      return null;
+    }
+  } else {
+    throw new Error('Headers or Header Name not provided.');
+  }
+};
+
+const trimLeadingSlashes = (str) => str.replace(/^\/+|\$/g, '');
+const trimTrailingSlashes = (str) => str.replace(/\/+$/g, '');
+
+const service = {
+  generateProxyHeaders: async (payload, currentUser, token) => {
+    if (!payload || !currentUser || !currentUser.idp) {
+      throw new Error('Cannot generate proxy headers with missing or incomplete parameters');
+    }
+
+    const headerData = {
+      formId: payload['formId'],
+      versionId: payload['versionId'],
+      submissionId: payload['submissionId'],
+      userId: currentUser.idpUserId,
+      username: currentUser.username,
+      firstName: currentUser.firstName,
+      lastName: currentUser.lastName,
+      fullName: currentUser.fullName,
+      email: currentUser.email,
+      idp: currentUser.idp,
+      token: token,
```

Comment:
The `token` is always being included, even when the external API is not allowed to receive the token. I realize that it would be more effort to make the check, but I think it would be better to only include the token when it will be used.

Reply:
Sure. Makes sense

Reply:
after thinking about it, we need to include all the information in this call. this is called on form load and we don't know what the API config is... (and the form could use multiple configs with different options).

can't implement this change.